docs/az-aro-python-development.md

```diff
@@ -0,0 +1,102 @@
+# How-to `az aro` development
+
+ARO-RP distributed the CLI command `az aro` to control the cluster. 
+Currently the `aro` subcommand is merged in the 
+[azure-cli](https://github.com/azure/azure-cli) code base and 
+is present as a command.
+
+Currently, there are two codebases for `aro` subcommand:
+
+* [upstream](https://github.com/Azure/ARO-RP/tree/master/python/az/aro) in main ARO-RP repo - which is an `az` extension
+* [downstream][https://github.com/Azure/azure-cli/tree/dev/src/azure-cli/azure/cli/command_modules/aro] - which is an `az` command
+
+Command is distributed alongside `az` and is automaticaly present in Azure cloud shell. Extension is installed as user action.
+
+Currently code have to be in sync for upstream and downstream.
+
+## Set up dev environment `az` for `aro` command
+
+Python virtual environment is required:
+
+```
+python3 -m venv env
+source <az dir>/env/bin/activate
+```
+
+in the `az` project directory works the best in case of command development.
+
+To work with `az` helper tool `azdev` comes in handy
+
+```
+pip install azdev
+```
+
+`azdev` tool **runs only** in the virtual environment.
+
+Then run
+
+```
+azdev setup -c
+```
```

Comment:
Added comment. I think this is both. When working in `azure-cli` they recomend to use separete virtual env. I think it up to developer to decide configuration (use our viirtual env) or create new one.

Reply:
ok, I'm looking for a default simple flow that makes it easier to onboard people.  If we can build this around something like `make pyenv`, that'd be good.

Reply:
For practical purposes, the `venv` module does the same as `virtualenv`. There are benefits to using it. It is part of core python, therefore it does not require an additional package. It is able to use python internals and does not have to do any tricks.

As far as I can understand the `make pyenv`, they can be synchronized.

I would do this in the follow-up PR.
However, this requires further thinking. Since there are now two flows

1) the code in ARO-RP, which is for the extension
2) the code in azure-cli, which is for the subcommand

They differ slightly, and the flow of work need to be aligned carefully to introduce the leat amount of confusion.